dotnet-source-8.yaml

```diff
@@ -0,0 +1,210 @@
+          - if: ${{build.arch}} == 'x86_64'
+            runs: |
+              tar zxf artifacts/x64/Release/dotnet-sdk-*.tar.gz -C "${{targets.destdir}}"/usr/share/dotnet
+      - runs: |
+          ln -s /usr/share/dotnet/dotnet "${{targets.destdir}}"/usr/bin/dotnet
+
+  - uses: strip
+
+subpackages:
+  - name: dotnet-8-runtime
+    description: "The .NET Core Runtime, version 8"
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}"/usr/share/dotnet/shared
+          mv "${{targets.destdir}}"/usr/share/dotnet/shared/Microsoft.NETCore.App "${{targets.subpkgdir}}"/usr/share/dotnet/shared/
+    dependencies:
+      runtime:
+        - dotnet-8
+
+  - name: dotnet-8-runtime-default
+    dependencies:
+      runtime:
+        - dotnet-8-runtime
+      provides:
+        - dotnet-runtime=8
+
+  - name: aspnet-8-runtime
+    description: "The ASP.NET Core Runtime, version 8"
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}"/usr/share/dotnet/shared
+          mv "${{targets.destdir}}"/usr/share/dotnet/shared/Microsoft.AspNetCore.App "${{targets.subpkgdir}}"/usr/share/dotnet/shared/
+    dependencies:
+      runtime:
+        - dotnet-8-runtime
+
+  - name: aspnet-8-runtime-default
+    dependencies:
+      runtime:
+        - aspnet-8-runtime
+      provides:
+        - aspnet-runtime=8
+
+  - name: dotnet-8-targeting-pack
+    description: "The .NET Core targeting pack, version 8"
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}"/usr/share/dotnet/packs
+          mv "${{targets.destdir}}"/usr/share/dotnet/packs/NETStandard.Library.* "${{targets.subpkgdir}}"/usr/share/dotnet/packs/
+          mv "${{targets.destdir}}"/usr/share/dotnet/packs/Microsoft.NETCore.App.* "${{targets.subpkgdir}}"/usr/share/dotnet/packs/
+
+  - name: aspnet-8-targeting-pack
+    description: "The ASP.NET targeting pack, version 8"
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}"/usr/share/dotnet/packs
+          mv "${{targets.destdir}}"/usr/share/dotnet/packs/Microsoft.AspNetCore.App.* "${{targets.subpkgdir}}"/usr/share/dotnet/packs/
+    dependencies:
+      runtime:
+        - dotnet-8-targeting-pack
+
+  - name: dotnet-8-sdk
+    description: "The .NET Core SDK, version 8"
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}"/usr/share/dotnet
+
+          for i in sdk sdk-manifests templates; do
+            mv "${{targets.destdir}}"/usr/share/dotnet/$i "${{targets.subpkgdir}}"/usr/share/dotnet/
+          done
+    dependencies:
+      runtime:
+        - dotnet-8-runtime
+        - aspnet-8-runtime
+        - dotnet-8-targeting-pack
+        - aspnet-8-targeting-pack
+
+  - name: dotnet-8-sdk-default
+    dependencies:
+      runtime:
+        - dotnet-8-sdk
+        - dotnet-8-runtime-default
+        - aspnet-8-runtime-default
+      provides:
+        - dotnet-sdk=8
```

Comment:
How come this builds both runtime & sdk? Isn't SDK somewhere separate at a much higher version?

My expectation for this package to build runtime only, i.e. 8.0.10 like what everyone else has.

And then some other .yaml build would be doing sdk build to get us up to higher version numbers. as in 8.0.403

Where does .NET SDK 8.0.403 come from, the one that PowerShell releases claim to use?

Reply:
The https://github.com/dotnet/dotnet is a source based monolithic repository which aggregates all: sdk, runtime, aspnet. It is pinning the versions of the all 3 elements (and more) to have a unified working matrix.

The .NET SDK 8.0.403 comes from the https://github.com/dotnet/installer/tree/v8.0.403 / https://github.com/dotnet/sdk/tree/v8.0.403, and the PowerShell version uses that version independently of the dotnet/dotnet latest matrix.

The .NET SDK 8.0.403 has not been yet released (picked up) into the monolithic repository, and that is why the latest PowerShell version cannot be built with the SDK from the dotnet/dotnet repository.

From what I understand, it is just a matter of time until that version gets picked up in the monolith.

An option would be to not use the monolithic repository and try to mix and match versions of the subprojects, but the downside would be that there will be less testing (which is done in the dotnet/dotnet), thus less reliable composition.
We would have to do the work of the maintainers / CI of dotnet/dotnet repo to make sure that our matrix works well.

For example, the last commit of the SDK present in the dotnet/dotnet repository (branch release/8.0.1xx), was
https://github.com/dotnet/sdk/commit/d90806d013db39755cda355f42015d6a248456f5, equivalent to SDK version v8.0.110.

It is a matter of choice, depending on how much upstream testing is deemed good-enough.

Reply:
THANKS! Let me ponder all that. I do think we need/want monolithic build for the bootstrap - aka known good point / LTS like thing. That makes all the things available.

And then it's likely on chainguard to build version-streamed non-monolithic sdk/aspnet/powershell etc.